### PR TITLE
Laboratory work #4, Kapitolina Untila - 21FPL1

### PR DESCRIPTION
**Pull request by KapitolinaVU**

### CODE REVIEW

File: lab_3_keywords_textrank/main.py [hunk 1]
```diff
@@ -4,22 +4,13 @@
...
 from typing import Optional, Union
-
 import csv
 
-from lab_1_keywords_tfidf.main import (
-    calculate_frequencies,
-    calculate_tf,
-    calculate_tfidf
-)
+from lab_1_keywords_tfidf.main import (calculate_frequencies, calculate_tf, calculate_tfidf, get_top_n)
```
Comment (artyomtugaryov) [inline lab_3_keywords_textrank/main.py:9]: The PR name is about the 4 lab. But it is implementation the 3 lab.

File: lab_4_summarization_textrank/main.py [hunk 2]
```diff
@@ -20,6 +20,7 @@ def __init__(self, text: str, position: int) -> None:
         """
         Constructs all the necessary attributes
         """
+
```
Comment (artyomtugaryov) [inline lab_4_summarization_textrank/main.py:23]: ?

File: lab_3_keywords_textrank/start.py [hunk 3]
```diff
@@ -3,20 +3,12 @@
 """
 
 from pathlib import Path
-from time import process_time
+from string import punctuation
```
Comment (demid5111) [inline lab_3_keywords_textrank/start.py:6]: discard changes

File: lab_3_keywords_textrank/target_score.txt [hunk 4]
```diff
@@ -1 +1 @@
-8
+10
```
Comment (demid5111) [inline lab_3_keywords_textrank/target_score.txt:1]: discard changes

File: lab_4_summarization_textrank/main.py [hunk 5]
```diff
@@ -20,59 +21,76 @@ def __init__(self, text: str, position: int) -> None:
...
         Constructs all the necessary attributes
         """
-        pass
+        if not isinstance(text, str):
+            raise ValueError
+        self._text = text
+        if not isinstance(position, int) or isinstance(position, bool):
+            raise ValueError
+        self._position = position
+        self._preprocessed: tuple[str, ...] = ()
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:30]: ?

File: lab_4_summarization_textrank/main.py [hunk 6]
```diff
@@ -84,59 +102,94 @@ def __init__(self, stop_words: tuple[str, ...], punctuation: tuple[str, ...]) ->
...
     def _split_by_sentence(self, text: str) -> tuple[Sentence, ...]:
         """
         Splits the provided text by sentence
         :param text: the raw text
         :return: a sequence of sentences
         """
-        pass
+        if not isinstance(text, str):
+            raise ValueError
+        splited_text = re.split(r'(?<=[.!?])\s+(?=[A-ZА-Я])', text.replace('\n', ' ').replace('  ', ' '))
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:125]: only one space?

File: lab_4_summarization_textrank/main.py [hunk 7]
```diff
@@ -84,59 +102,94 @@ def __init__(self, stop_words: tuple[str, ...], punctuation: tuple[str, ...]) ->
...
         :return:
         """
-        pass
+        if not isinstance(tokens, tuple):
+            raise ValueError
+        for token in tokens:
+            if not isinstance(token, str):
+                raise ValueError
+        my_tokens = (token for token in tokens if token not in self._word2id)
+        for index, token in enumerate(my_tokens, start=1000 + len(self._word2id)):
```
Comment (demid5111) [inline lab_4_summarization_textrank/main.py:176]: own attribute of instance and it should be mutated

**Review by demid5111 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**